rundeckapp/grails-spa/packages/ui-trellis/src/app/pages/project-nodes-config/ProjectConfigurableForm.vue

```diff
@@ -0,0 +1,167 @@
+<template>
+  <div>
+    <div class="form-horizontal">
+        <plugin-config
+            v-if="isConfigSet"
+            :mode="'create'"
+            :show-title="false"
+            :show-description="false"
+            :use-runner-selector="true"
+            v-model="finalExtraConfig"
+            :event-bus="rundeckContext.eventBus"
+            :plugin-config="pluginConfig"
+            :category="category"
+        >
+          <template v-slot:extra>
+            <div class="row">
+              <div class="col-xs-12 col-sm-12">
+                    <span>
+                        <a
+                            class="btn btn-cta reset_page_confirm"
+                            @click="saveConfig"
+                            :key="'save'"
+                        >{{"Save"}}</a>
+                    </span>
+              </div>
+            </div>
+          </template>
+        </plugin-config>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import PluginConfig from "../../../library/components/plugins/pluginConfig.vue"
+import {getRundeckContext, RundeckContext} from "../../../library";
+import {getProjectConfigurable, setProjectConfigurable} from "./nodeSourcesUtil";
+import {Notification} from "uiv"
+import {defineComponent, onMounted, ref} from "vue";
+interface ConfigurableItem{
+  name?:string
+  properties?:[{}]
+  propertiesMapping?:any
+  values?:any
+}
+
+
+export default defineComponent({
+  name: "ProjectConfigurableForm",
+  components: {
+    PluginConfig,
+  },
+  data() {
+    return {
+      rundeckContext: {} as RundeckContext,
+      pluginConfig: {},
+      extraConfigSet: {},
+      isConfigSet: false,
+      projectConfigurableNames: [],
+      finalExtraConfig: {config: {}},
+      originalConfig: {},
+    };
+  },
+  props: ['category', 'categoryPrefix', 'helpCode', 'serviceName'],
+  methods: {
+    async loadConfig() {
```

Comment:
suggestion: I am aware that you kept the name extraConfigSet from the original code in the gsp, but the code will become a little bit cleaner if you rename and tweak the data a little bit, because it's hard to infer without further reading, what is the difference between extraConfigSet, the finalExtraConfig, the originalConfig and which one is being mutated by user input.

So this is my suggestion, but feel free to name it differently:

extraConfigSet -> apiResponse or fetchedData
finalExtraConfig -> cache (It would be great if there wasn't the need for having a property config nested, but saw that it's how pluginConfig works)
originalConfig -> massagedConfig

Also, would consider to have originalConfig as a local variable instead of living in the data and to move lines 68-92 to its own method (or to add a comment before pointing out that you're massaging the data);